### DISCUSSION TITLE
RStudio quarto preview "shaking" after knitting

### DISCUSSION THREAD

**Discussion by shamindras** [quarto-dev/quarto-cli#3104] [Q&A] [answered]
I'm currently trying to knit and iteratively develop my [quarto website](https://www.shamindras.com/) (source [here](https://github.com/shamindras/ss_personal_quarto_blog)). 

When I knit only the main [`index.qmd`](https://github.com/shamindras/ss_personal_quarto_blog/blob/main/index.qmd) welcome page, RStudio viewer starts "shaking".
I've recorded a `gif` below to illustrate the issue.

![CleanShot 2022-10-29 at 17 24 44](https://user-images.githubusercontent.com/7627188/198853052-11ca5f98-3a53-4fb3-947a-795b65fb6a57.gif)

It only does this for this one page, all other `qmd` pages do not shake like this.

I'm using a relatively fast machine with latest R, RStudio, and quarto (see specs below). I've tried restarting the entire computer, but this still persists. I'm not sure what changed to cause this. Did not have the issue until this morning.

Could anyone please help resolve the issue? I'm getting quite dizzy editing the website 😄.

<details>
<summary>Specs</summary>
Note: I'm running macOS with the following specs

![CleanShot 2022-10-29 at 17 31 57@1x](https://user-images.githubusercontent.com/7627188/198853176-60f0af3c-ba49-45cb-8641-15c6f5ecde4a.png)

**RStudio version:** 2022.07.2

**R version:** 4.2.1

**quarto version:** 1.1.251
</details>

**Comment by mcanouil** [comment]
May I suggest to check if it is already reported in https://github.com/rstudio/rstudio/issues and report it if this is not.
It seems to be an RStudio issue.


The issue seems to be about the vertical scrolling bar being added or not in the viewer.

**Reply by shamindras** [reply]
@mcanouil - sure, sounds good. I just opened up the issue at rstudio/rstudio#12255.

**Reply by shamindras** [reply]
@mcanouil - just as an update, this issue was resolved per the fix proposed [here](https://github.com/rstudio/rstudio/issues/12255#issuecomment-1300962503) by @ronblum.

Thought to let the `quarto` community know in case others have the same issue.

**Reply by mcanouil** [reply]
Thanks for the update.
Maybe switch the category from "General" to "Q&A" and write as a new answer the workaround and the link to the issue you opened, then mark you new answer as the answer to let others know.

PS: No need to ping (mention) me, especially when replying.

**Reply by shamindras** [reply]
sure - done, see below.

Note: I pinged you as a general courtesy, to thank you for kindly directing me to file an issue with RStudio. I'll make sure to not ping in replies (unless necessary) going forward.

Thanks for your help on this.

**Comment by shamindras** [comment, marked as answer]
As [requested](https://github.com/quarto-dev/quarto-cli/discussions/3104#discussioncomment-4040485) for an update to this issue, this is indeed a scroll bar bug in the RStudio IDE which will hopefully get resolved soon per [this issue](https://github.com/rstudio/rstudio/issues/12255) I'd opened.

There is a temporary macOS fix per the discussion here, which can be summarized from my comment [here](https://github.com/rstudio/rstudio/issues/12255#issuecomment-1300962503) (thanks to @ronblum) as follows:

> ...I changed my `System Preferences -> General -> Show Scroll Bars` from `Always` to `When scrolling`. That is, I have updated to the following setting:
> 
> ![CleanShot 2022-11-02 at 13 19 13@2x](https://user-images.githubusercontent.com/7627188/199557957-5239c7af-37fd-4e96-bf32-fb4a9d980ede.png)
> 
> The "shaking" effect stopped.
> 

Hope this helps others who have similar RStudio/quarto issues on macOS in the meantime.
